### PR TITLE
Heatmap

### PR DESCRIPTION
**Pull request by vocosu**
Finished heatmap.

References https://github.com/odota/mobile/issues/55.

### CODE REVIEW

**Comment by Proyoyo** [comment]
@vocosu Everything looks great on Android. However, on iOS, there's some scaling issues.

asset

asset 1

As you can see, this only happens to several player. I tested this on both iPhone 5S and iPhone 6s+. Is it possible for you to try and debug this in Xcode?


In addition, we should probably provide different map images based on the patch version. This is already implemented in the ui repo.

**Comment by vocosu** [comment]
For your first question, I don't have a Mac or iOS device so might take me a little longer to solve that issue. I'll try to set up a VM this week so I can start debugging in Xcode.

I can definitely add the different map images based on the patch version though.

**Comment by Proyoyo** [comment]
@vocosu Gotcha. And don't forget to pull from master before submitting the PR. I have updated master with 7.07 compatibility.

I'll spend most of my time this week fixing the bugs that were found by the open beta testers (e.g. iPhone X compatibility, custom font issue). Once I'm done with that, I can help you with the iOS debugging.

**Comment by vocosu** [comment]
@Proyoyo  I have implemented changing the map image based on the patch version.

Still stuck on the iOS bug due to issues setting up my VM. I'll have access to an iOS device this weekend though. Let me know when you can help me on the iOS debugging.

**Comment by Proyoyo** [comment]
Looks good! I can probably help with iOS debugging later today or tomorrow.

File: app/components/Heatmap.js [hunk 1]
```diff
@@ -8,17 +9,65 @@ import { connect } from 'react-redux';
...
 
     render() {
         let webview = null;
         if(this.props.points) {
-            console.log(this.props.points);
             uri = Platform.OS === 'ios' ? 'heatmap.html' : 'file:///android_asset/heatmap.html';
-            script = heatmapInputGenerator(this.props.points, 300);
+            const width = 285;
+            const adjustedData = this.scaleAndExtrema(this.props.points, width / 127, null, 25)
+            script = heatmapInputGenerator(JSON.stringify(adjustedData), width, this.getMap(this.props.starTime));
```
Comment (Proyoyo) [inline app/components/Heatmap.js]: This should be this.props.startTime

File: app/containers/MatchLaning.js [hunk 2]
```diff
@@ -213,11 +213,15 @@ class MatchLaning extends Component {
...
-        //<Heatmap points = {rowData.unpackedPosition} background = {bgColor}/>
         if(toggled) {
+            console.log(this.props)
             additionalInfo = (
                 <View style = {additionalRowContainer}>
-
+                    <Heatmap 
+                        points = {rowData.unpackedPosition} 
+                        background = {bgColor} 
+                        startTime={this.props.matchDetails.startTime}
```
Comment (Proyoyo) [inline app/containers/MatchLaning.js]: This should be change to this.props.matchDetails.start_time

**Review by Proyoyo — CHANGES_REQUESTED** [review]
The map based on game start time doesn't work. I've added a few comments with the instruction on how to fix it.

File: app/containers/MatchLaning.js (hunk 2 above)
Comment (vocosu) [reply app/containers/MatchLaning.js]: Good catch. Just squashed the fix into the same commit.

File: app/components/Heatmap.js (hunk 1 above)
Comment (vocosu) [reply app/components/Heatmap.js]: Fixed.

**Comment by Proyoyo** [comment]
I've tried it on iOS and now the image doesn't get rendered anymore. I've removed the XCode reference to detailed.png and added new XCode references to the new map images. However, this doesn't solve the issue. I'll investigate this further tomorrow.

asset 2

**Comment by vocosu** [comment]
**Edit (12-20-2017)**: If someone else wants to take a stab at this please do so. I don't have full access to a Mac so I can't debug this.

**Comment by anhmiuhv** [comment]
I made a bug fixes in this branch
#101

**Comment by vocosu** [comment]
Closing due to https://github.com/odota/mobile/pull/101.

**Pull request closed without merging**